graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -0,0 +1,342 @@
+        default=None,
+        help=(
+            "The desired number of output files to be crea ted per type. "
+            "If set to '', None, or '-1', we let Spark decide."
+        ),
+    )
+    parser.add_argument(
+        "--add-reverse-edges",
+        type=lambda x: (str(x).lower() in ["true", "1"]),
+        default=True,
+        help="When set to True, will create reverse edges for every edge type.",
+    )
+
+    parser.add_argument(
+        "--log-level",
+        type=str,
+        default="INFO",
+        help="Logging level, default is INFO",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    """Main entry point for GSProcessing"""
+    # Allows us to get typed arguments from the command line
+    gsprocessing_args = GSProcessingArguments(**vars(parse_args()))
+    logging.basicConfig(level=gsprocessing_args.log_level)
+
+    # Determine if we're running within a SageMaker container
+    is_sagemaker_execution = os.path.exists("/opt/ml/config/processingjobconfig.json")
+
+    if gsprocessing_args.input_prefix.startswith("s3://"):
+        assert gsprocessing_args.output_prefix.startswith(
+            "s3://"
+        ), "When providing S3 input and output prefixes, they must both be S3."
+        filesystem_type = "s3"
+    else:
+        # Ensure input and output prefixes exist and convert to absolute paths
+        gsprocessing_args.input_prefix = str(
+            Path(gsprocessing_args.input_prefix).resolve(strict=True)
+        )
+        if not Path(gsprocessing_args.output_prefix).absolute().exists():
+            os.makedirs(gsprocessing_args.output_prefix)
+        gsprocessing_args.output_prefix = str(
+            Path(gsprocessing_args.output_prefix).resolve(strict=True)
+        )
+        filesystem_type = "local"
+
+    # local input location for config file and execution script
+    if is_sagemaker_execution:
+        local_config_path = "/opt/ml/processing/input/data"
+    else:
+        # If not on SageMaker, assume that we are running in a
+        # native env with local input or Docker execution with S3 input
+        if filesystem_type == "local":
+            local_config_path = gsprocessing_args.input_prefix
+        else:
+            tempdir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+            input_bucket = gsprocessing_args.input_prefix.split("/")[2]
+            input_s3_prefix = gsprocessing_args.input_prefix.split("/", 3)[3]
+            s3 = boto3.client("s3")
+            s3.download_file(
+                input_bucket,
+                f"{input_s3_prefix}/{gsprocessing_args.config_filename}",
+                os.path.join(tempdir.name, gsprocessing_args.config_filename),
+            )
+            local_config_path = tempdir.name
+
+    # local output location for metadata files
+    if is_sagemaker_execution:
+        local_output_path = "/opt/ml/processing/output"
```

Comment:
Why not store files under /tmp/ ?

Reply:
This is where we store output JSON files that we want to be persisted/uploaded to S3 after the job finishes.

The naming here is a bit confusing, but I'd rather leave a re-design for 0.3. The structure is a bit of remnant from our previous implementation.